goldenmaster/Plugins/ApiGear/Source/ApiGearOLink/Private/OLinkHostConnection.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+THIRD_PARTY_INCLUDES_START
+#include "olink/remoteregistry.h"
+#include "olink/remotenode.h"
+THIRD_PARTY_INCLUDES_END
+#include <UObject/Object.h>
+#include "ApiGearLogCategories.h"
+#include "INetworkingWebSocket.h"
+
+/// @brief an instance of this class is held by the OLink server worker for each existing connection
+class APIGEAROLINK_API FOLinkHostConnection
+{
+
+public:
+	FOLinkHostConnection(INetworkingWebSocket* InSocket, ApiGear::ObjectLink::RemoteRegistry& InRegistry, ApiGear::ObjectLink::WriteLogFunc logFunc)
+		: Socket(InSocket)
+		, Registry(InRegistry)
+		, Node(ApiGear::ObjectLink::RemoteNode::createRemoteNode(InRegistry))
+	{
+		Node->onLog(logFunc);
+		Node->onWrite([InSocket](std::string msg)
+			{
+				check(InSocket);
+				InSocket->Send(reinterpret_cast<const uint8*>(msg.c_str()), msg.size(), false);
```

Comment:
I'd extract function encode or something like that.
It should be somehow assured that encoding and decoding is done in a same way.  Decoding is done by UnrealOLinkHostWorker. Maybe it can use an object - message converter that it would pass here?

Reply:
We pass the data to the websocket or receive from there.
Since we can't change the API we can't pass an object for this purpose.
We can only hide some of the casting in a local function.
Not sure whether that's any better because we can't ensure that en/decoding are done the same way.